src/stacked_borrows.rs

```diff
@@ -80,16 +78,33 @@ impl fmt::Debug for Item {
     }
 }
 
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
+pub enum ItemOrUnknown {
+    Item(Item),
+    Unknown(PtrId),
```

Comment:
Sine you have `unknown_bottom`, what is this for?

Reply:
Using a `Vec<ItemOrUnknown>` will probably make the items in that vec bigger than they used to be, which is a problem since Stacked Borrows already leads to enormous memory consumption for big stacks. We should really avoid growing the type of elements of that `Vec`.